sample/github/README.md

```diff
@@ -6,156 +6,130 @@ deployed to receive the webhook's event deliveries and forward them into a
+Kubernetes resources. In a production environment, you might want to limit the
+access of this service account to only specific namespaces.
 
 ```shell
- ko apply -f sample/github/githubsecret.yaml
+kubectl apply -f eventing/samples/github-events/auth.yaml
 ```
 
-Then create a PR for the repo you configured the webhook for, and you'll see that the Title
-will be modified with the suffix '(looks pretty legit)'
+## Build and deploy the sample
+
+1.  Use Docker to build the sample code into a container. To build and push with
+    Docker Hub, run these commands replacing `{username}` with your Docker Hub
+    username. Run the following from the _root_ of the `knative/docs` repo:
+
+    ```shell
+    # Build the container on your local machine
+    docker build -t {username}/github-events --file=eventing/samples/github-events/Dockerfile .
+
+    # Push the container to docker registry
+    docker push {username}/github-events
+    ```
+
+1.  After the build has completed and the container is pushed to Docker Hub, you
+    can deploy the function into your cluster. **Ensure that the container image
+    value in `function.yaml` matches the container you built in the previous
+    step.** Apply the configuration using `kubectl`:
+
+    ```shell
+    kubectl apply -f eventing/samples/github-events/function.yaml
+    ```
+
+1.  Check that your service is running using:
+
+    ```shell
+    kubectl get services.serving.knative.dev -o "custom-columns=NAME:.metadata.name,READY:.status.conditions[2].status,REASON:.status.conditions[2].message"
+    NAME              READY     REASON
+    legit             True      <none>
+    ```
+
+1.  Create a [personal access token](https://github.com/settings/tokens) to 
+    GitHub repo that the GitHub source can use to register webhooks with the 
+    GitHub API. Also decide on a token that your code will use to authenticate
+    the incoming webhooks from GitHub (*accessToken*). Update 
+    `eventing/samples/github-events/githubsecret.yaml` with those values. If 
+    your generated access token is `'asdfasfdsaf'` and you choose your
+    *secretToken* as `'password'`, you'd modify 
+    `eventing/samples/github-events/githubsecret.yaml` like so:
+    
+    ```yaml
+    apiVersion: v1
+    kind: Secret
+    metadata:
+      name: githubsecret
+    type: Opaque
+    stringData:
+      githubCredentials: >
+        {
+          "accessToken": "asdfasfdsaf",
+          "secretToken": "password"
+        }
+    ```
+    
+    Then, apply the githubsecret using `kubectl`:
+    
+    ```shell
+    kubectl apply -f eventing/samples/github-events/githubsecret.yaml
+    ```
+
+1.  Create the flow sending GitHub Events to the service:
+
+    ```shell
+    kubectl apply -f eventing/samples/github-events/flow.yaml
+    ```
+
+1.  Create a PR for the repo you configured the webhook for, and you'll see that
+    the Title will be modified with the suffix `(looks pretty legit)`
+
+
+## Understanding what happened
+
+...TODO...
```

Comment:
I'd make this more formal:

**TODO:** explain the resources and communication channels, as well as where the secret is used.

Reply:
In particular include a note to look at `https://github.com/<owner>/<repo>/settings/hooks` to see the webhook registered and then deleted.